addons/partner_autocomplete/models/res_partner.py

```diff
@@ -124,12 +126,21 @@ def enrich_company(self, company_domain, partner_gid, vat, timeout=15):
 
     @api.model
     def read_by_vat(self, vat, timeout=15):
+        check_vies_res = check_vies(vat)
+        del check_vies_res['requestDate']
+        vies_result = {res: check_vies_res[res] for res in check_vies_res}
         vies_vat_data, _ = self.env['iap.autocomplete.api']._request_partner_autocomplete('search_vat', {
             'vat': vat,
+            'vies_result': vies_result,
         }, timeout=timeout)
         if vies_vat_data:
             return [self._format_data_company(vies_vat_data)]
         else:
+            if vies_result.get('valid'):
+                return [{
+                    'name': vies_result.get('name'),
```

Comment:
For some country (e.g. Germany), VIES won't return the name of the company, it will return `---` instead, this case should be handled.